src/app/Hooks/components/HooksTable.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import { Button, Flex, FlexItem, Level, LevelItem, Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  IRow,
+  sortable,
+  wrappable,
+  classNames,
+  cellWidth,
+} from '@patternfly/react-table';
+import alignment from '@patternfly/react-styles/css/utilities/Alignment/alignment';
+
+import { useSortState, usePaginationState } from '@app/common/hooks';
+import { IHook } from '@app/queries/types';
+import { FilterToolbar, FilterType, FilterCategory } from '@app/common/components/FilterToolbar';
+import { useFilterState } from '@app/common/hooks/useFilterState';
+import TableEmptyState from '@app/common/components/TableEmptyState';
+import HookActionsDropdown from './HookActionsDropdown';
+
+interface IHooksTableProps {
+  hooks: IHook[];
+  openCreateHookModal: () => void;
+  openEditHookModal: (hook: IHook) => void;
+}
+
+const HooksTable: React.FunctionComponent<IHooksTableProps> = ({
+  hooks,
+  openCreateHookModal,
+  openEditHookModal,
+}: IHooksTableProps) => {
+  const filterCategories: FilterCategory<IHook>[] = [
+    {
+      key: 'name',
+      title: 'Name',
+      type: FilterType.search,
+      placeholderText: 'Filter by name...',
+      getItemValue: (item) => {
+        return item.metadata.name;
+      },
+    },
+    {
+      key: 'url',
+      title: 'URL',
+      type: FilterType.search,
+      placeholderText: 'Filter by URL...',
+      getItemValue: (item) => {
+        return item.spec.url;
+      },
+    },
+    {
+      key: 'branch',
+      title: 'Branch',
+      type: FilterType.search,
+      placeholderText: 'Filter by branch...',
+      getItemValue: (item) => {
+        return item.spec.branch;
+      },
+    },
+  ];
+
+  const { filterValues, setFilterValues, filteredItems } = useFilterState(hooks, filterCategories);
+  const getSortValues = (hook: IHook) => [
+    hook.metadata.name,
+    hook.spec.url,
+    hook.spec.branch,
+    '', // plans column,
```

Comment:
If we are keeping this column (I assume we'd drop it if we drop the pass-by-reference behavior), we should leave a TODO so we remember to put a real sort value here (the number of associated plans). The values in that column will be a pain to get actually, we'll have to look them up from a plans query.

Reply:
Done